### ISSUE TITLE
'Invalid parameter count' thrown during setup of Callback for Instance methods that return void 

### ISSUE THREAD

**Issue by paulose** [vanderkleij/Smocks#11]
```C#
context.Setup(() => It.IsAny<MessageQueue>().Send(message)).Callback((Message m) =>
{
    called = true;
});
```

When calling the above line, Smocks throws an error 'Invalid parameter count'. I am trying to mock the System.Messaging.MessageQueue.Send method here.

**Comment by vanderkleij** [comment]
Thanks for reporting this problem. I can reproduce the issue and will look into it shortly. 

If you only intend to use this to confirm that the setup was called, you can use `.Verifiable()` instead of `Callback()`, followed by a call to `context.Verify()` after the method should've been invoked.

**Comment by paulose** [comment]
Thanks. I got it working for the moment with Verifiable. The issue seems to be in the condition for extraParameters 

protected int ParameterCount
        {
            get
            {
                int extraParameters = (!MethodCall.Method.IsStatic && 
                    !MethodCall.Method.IsConstructor) ? 1 : 0;
                return MethodCall.Method.GetParameters().Length + extraParameters;
            }
        }

**Comment by vanderkleij** [comment]
Fixed in [nuget package 0.4.17](https://www.nuget.org/packages/Smocks/0.4.17). Thanks for reporting and pointing out where the issue arose! :+1:

**Comment by paulose** [comment]
Works like a charm. Thanks for closing the issue at that speed.
